rai/node/bootstrap.cpp

```diff
@@ -1536,6 +1573,190 @@ request (std::move (request_a))
 	set_current_end ();
 }
 
+/**
+ * Bulk pull of a range of blocks, or a checksum for a range of
+ * blocks [min_hash, max_hash) up to a max of max_count.  mode
+ * specifies whether the list is returned or a single checksum
+ * of all the hashes.  The checksum is computed by XORing the
+ * hash of all the blocks that would be returned
+ */
+void rai::bulk_pull_blocks_server::set_params ()
+{
+	assert (request != nullptr);
+
+	if (connection->node->config.logging.bulk_pull_logging ())
+	{
+		std::string modeName = "<unknown>";
+
+		switch (request->mode)
+		{
+			case rai::bulk_pull_blocks_mode::list_blocks:
+				modeName = "list";
+				break;
+			case rai::bulk_pull_blocks_mode::checksum_blocks:
+				modeName = "checksum";
+				break;
+		}
```

Comment:
I left default out here and supplied a default value (on line 1589) so that if new modes are added that the compiler warns about the missing case here.

Reply:
I like this. I might even consider throwing a static assert false in the default case.

Reply:
Yeah this seems to be the better solution.  Default case would also be a good idea to add @rkeene

Reply:
Ok, I see, good idea.